### PR TITLE
Fix default return value of mocked __toString()

### LINKED ISSUE #267: Configuring stubbed __toString() does not work
There is a regression in phpunit-mock-objects 3.0 (used with PHPUnit 5) from phpunit-mock-objects 2.3 (used with PHPUnit 4):

```php
<?php
class Example
{
    public function __toString()
    {
    }
}

class Test extends PHPUnit_Framework_TestCase
{
    public function testOne()
    {
        $stub = $this->getMock(Example::class);
        $stub->method('__toString')->willReturn('string');

        $this->assertEquals('string', (string) $stub);
    }
}
```

## Expected
```
PHPUnit 5.0.8 by Sebastian Bergmann and contributors.

.                                                                   1 / 1 (100%)

Time: 23 ms, Memory: 3.25Mb

OK (1 test, 1 assertion)
```

## Actual
```
PHPUnit 5.0.8 by Sebastian Bergmann and contributors.

F                                                                   1 / 1 (100%)

Time: 28 ms, Memory: 3.75Mb

There was 1 failure:

1) Test::testOne
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'string'
+''

/home/sb/Test.php:16

FAILURES!
Tests: 1, Assertions: 1, Failures: 1.
```

### PR DESCRIPTION
**Pull request by danez**
Moved the logic of detecting if it is the toString() method to
Invocation/Object. I think it is better to check there as in my 
opinion fits better as all the other default return values are 
also generated there.

I also added a test to see that mocking of toString() works.

Do you think this change is okay?

Fixes #267

### CODE REVIEW

**Comment by cirpo** [comment]
:+1:  please lets merge and ship it :)

**Comment by p16** [comment]
:+1: thanks! :)

File: src/Framework/MockObject/Invocation/Object.php [hunk 1]
```diff
@@ -33,4 +33,16 @@ public function __construct($className, $methodName, array $parameters, $returnT
         parent::__construct($className, $methodName, $parameters, $returnType, $cloneObjects);
         $this->object = $object;
     }
+
+    /**
+     * @return mixed Mocked return value.
+     */
+    public function generateReturnValue()
+    {
+      if (strtolower($this->methodName) == '__tostring') {
```
Comment (nicolas-grekas) [inline src/Framework/MockObject/Invocation/Object.php]: please use four spaces indentation

File: tests/MockObjectTest.php [hunk 2]
```diff
@@ -841,4 +841,12 @@ public function testStringableClassDoesNotThrow()
 
         $this->assertInternalType('string', (string) $mock);
     }
+
+    public function testStringableClassCanBeMocked()
+    {
+        $mock = $this->getMock('StringableClass');
+        $mock->method('__toString')->willReturn('somestring');
+
+        $this->assertEquals('somestring', (string) $mock);
```
Comment (nicolas-grekas) [inline tests/MockObjectTest.php]: I'd say assertSame is stricter, thus better

File: src/Framework/MockObject/Invocation/Object.php [hunk 3]
```diff
@@ -33,4 +33,16 @@ public function __construct($className, $methodName, array $parameters, $returnT
         parent::__construct($className, $methodName, $parameters, $returnType, $cloneObjects);
         $this->object = $object;
     }
+
+    /**
+     * @return mixed Mocked return value.
```
Comment (nicolas-grekas) [inline src/Framework/MockObject/Invocation/Object.php]: `{@inheritdoc}`

**Comment by tim-bezhashvyly** [comment]
Any update on this?

**Comment by sebastianbergmann** [comment]
I do not have time to work on PHPUnit this week, I'm afraid.

**Comment by sebastianbergmann** [comment]
Should be fixed by b28b029356e65091dfbf8c2bc4ef106ffece509e.

**Pull request closed without merging**